scripts/build/TestImpactAnalysis/git_utils.py

```diff
@@ -32,7 +32,9 @@ def create_diff_file(self, src_commit_hash: str, dst_commit_hash: str, output_pa
 
         try:
             # Remove the existing file (if any) and create the parent directory
-            output_path.unlink(missing_ok=True)
+            # output_path.unlink(missing_ok=True) # missing_ok is only available in Python 3.8+
+            if output_path.is_file():
+                output_path.unlink()
```

Comment:
Unlink can raise an `OSError` as well so we should add an exception cause for that.

Reply:
discussed offline, the caller cannot recover from that exception due to the fact that a uuid is passed in as the filename so we are letting it propagate to be caught at the entry point and dumped out as an error msg and stack trace for user feedback.